Library/TestHelpers/MockAppTrackingTransparency.swift

```diff
@@ -1,19 +1,45 @@
 import Library
+import ReactiveSwift
 
 class MockAppTrackingTransparency: AppTrackingTransparencyType {
   public private(set) var advertisingIdentifier: String?
-  public var shouldRequestAuthStatus = true
   public var requestAndSetAuthorizationStatusFlag = false
 
+  public var shouldRequestAuthStatus: Bool {
+    get {
+      self.authorizationStatusProperty.value == .notDetermined
+    }
+    set {
+      self.authorizationStatusProperty.value = (newValue ? .notDetermined : .authorized)
+    }
+  }
+
+  public var authorizationStatusValue: AppTrackingAuthorization {
+    get {
+      self.authorizationStatusProperty.value
+    }
+    set {
+      self.authorizationStatusProperty.value = newValue
+    }
+  }
+
   func updateAdvertisingIdentifier() {
     self.advertisingIdentifier = self.shouldRequestAuthStatus ? "advertisingIdentifer" : nil
   }
 
   func requestAndSetAuthorizationStatus() {
+    self.authorizationStatusProperty.value = .authorized
     self.advertisingIdentifier = self.requestAndSetAuthorizationStatusFlag ? "advertisingIdentifier" : nil
   }
 
   func shouldRequestAuthorizationStatus() -> Bool {
     self.shouldRequestAuthStatus
   }
+
+  private var authorizationStatusProperty = MutableProperty<AppTrackingAuthorization>(.notDetermined)
+  var authorizationStatus: SignalProducer<AppTrackingAuthorization, Never> {
+    self.authorizationStatusProperty.producer
+      .replayLazily(upTo: 1)
```

Comment:
how does this work? I've never seen .replayLazily

Reply:
The behavior is that when someone observes the values from this observable thing, they should get the current authorization value immediately and then any changes later.

With ReactiveSwift, a Signal is "hot", meaning that the pipeline only runs once, and everyone observing it gets the same values at the same time. That means that you can't have a "return the current value immediately" behavior using a Signal, because observing doesn't start anything, and since every observer gets the same values, subscribing anywhere would need to re-emit the current value to all the other observers who wouldn't need it.

Instead this is a SignalProducer, which is "cold", meaning that the pipeline doesn't start until you observe it, and every observation gets a separate pipeline. This means you CAN have a "return the current value immediately" behavior, because when you observe a SIgnalProducer, it creates a Signal, making it "hot" for the lifetime of its Disposable, and that can emit a value immediately.

Sometimes you want a hybrid of these approaches, where one SignalProducer acts like a Signal and runs a single pipeline with values shared to multiple observers, but where you can also modify the pipeline for each individual observer. `replayLazily(upTo:)` does this by buffering up to a certain number of values, returning those immediately to new observers. And then any values after that are shared to all observers.

So that's what's going on here. When someone observes this initially, they get the current buffered value from the property, and then any subsequent values.